addons/delivery/models/stock_picking.py

```diff
@@ -55,6 +55,20 @@ def _compute_weight_uom_name(self):
 class StockPicking(models.Model):
     _inherit = 'stock.picking'
 
+    def _auto_init(self):
+        if not column_exists(self.env.cr, "stock_picking", "weight"):
+            # In order to speed up module installation when dealing with hefty data
+            # We create the column weight manually, but the computation will be skipped
+            # Therefore we do the computation in a query by getting weight sum from stock moves
+            create_column(self.env.cr, "stock_picking", "weight", "NUMERIC")
+            self.env.cr.execute("""
+                with compute_weight as (
+                    select sum(weight) as weight_sum, picking_id
+                    from stock_move group by picking_id)
+                update stock_picking picking
+                set weight = (select weight_sum from compute_weight where picking_id = picking.id);
+            """)
```

Comment:
I'm a fan of the SQL code writing standard :nerd_face: Also, may as well exclude the irrelevant moves. Should probably be double checked that I didn't mess up anything with my suggested changes.

```suggestion
                WITH computed_weight AS (
                    SELECT SUM(weight) AS weight_sum,
                           picking_id
                    FROM stock_move
                    WHERE picking_id IS NOT NULL
                    GROUP BY picking_id
                )
                UPDATE stock_picking
                SET weight = weight_sum
                FROM computed_weight
                WHERE stock_picking.id = computed_weight.picking_id;
            """)
```

Reply:
Nice catch! I didn't think about moves with no `picking_id` ^^